src/NSCBC_outflow.hpp

```diff
@@ -35,6 +35,12 @@ AMREX_GPU_DEVICE AMREX_FORCE_INLINE auto dQ_dx_outflow(quokka::valarray<amrex::R
 	const amrex::Real w = Q[3];
 	const amrex::Real P = Q[4];
 
+	static constexpr int nmscalars_ = Physics_Traits<problem_t>::numMassScalars;
+	amrex::GpuArray<Real, nmscalars_> massScalars;
+	for (int n = 0; n < nmscalars_; ++n) {
+		massScalars[n] = Q[RadSystem<problem_t>::scalar0_index + n];
```

Comment:
@BenWibking  Is it correct to access the indices of scalars this way?

If yes, we should also use this way to specify the index of density, etc. above, which are currently hardcoded.

Reply:
Here they follow the ordering defined by this function:
https://github.com/quokka-astro/quokka/blob/9d44f9156beba9a5e4b91ce460b4323f2e1663c0/src/hydro_system.hpp#L322

This is not necessarily the same as the ordering used by the HydroSystem/RadSystem indices, although in the current implementation they are the same.